src/main/jbake/content/blog/2017/07/grafana-new-query-interface.adoc

```diff
@@ -0,0 +1,174 @@
+= Grafana: new query interface
+Joel Takvorian
+2017-07-20
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, metrics, grafana
+:figure-caption!:
+
+There has been improvements lately in our Hawkular Grafana datasource, that are worth mentioning.
+
+- The way to query metrics by tags has changed since plugin link:https://github.com/hawkular/hawkular-grafana-datasource/releases/tag/v1.0.8[v1.0.8].
+It now takes advantage of the Hawkular Metrics' tags query language,
+that was introduced server-side in link:http://www.hawkular.org/blog/2017/02/08/hawkular-metrics-0.24.0.Final-released.html[Metrics 0.24.0]
+and enhanced in link:http://www.hawkular.org/blog/2017/03/07/hawkular-metrics-0.25.0.Final-released.html[0.25.0].
+To sum it up, Metrics integrates a parser that allows queries such as:
+`tag1 = 'awesome!' AND tag2 NOT IN ['foo', 'bar']`.
+
+- The datasource in now able to fetch bucketized metrics stats, instead of raw metrics.
+It consists in aggregating datapoints in slices of time (buckets) and providing, for each slice, some statistics like _min_, _max_, _average_ and more.
+The exact content of a bucket is described in link:http://www.hawkular.org/docs/rest/rest-metrics.html#NumericBucketPoint[Metrics REST API].
+Hawkular has always been able to provide them server-side,
+but being able to use them in the Grafana plugin is new, introduced in link:https://github.com/hawkular/hawkular-grafana-datasource/releases/tag/v1.0.9[v1.0.9].
+
+[.text-center]
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-20-grafana-empty-query.png[New query interface]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-empty-query.png[New query interface]
+endif::[]
+
+[.text-center]
+_The new query interface_
+
+=== Tags query language
+
+The first change is that you don't have to choose between _query by tag_ and _query by metric id_ anymore.
+You can mix them, _ie._ start querying by tag and that will refine the available list of metric names, acting as a filter.
```

Comment:
I'm not totally convinced on this one, it makes several repetitions. What about totally rephrased:

The first change is that you don't have to choose between _query by tag_ and _query by metric id_ anymore, you can do both at the same time.
Querying by tag will refine the available list of metric names (much like a filter) and can result in multiple metrics from a single query.
By selecting a _metric name_, you restrict the query to only display that one.